src/encDecAlgo.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const narsissticNumber = {};                // {'0': 1, '1': 1,...., '10': 153, '11': 370}
+
+function getCharCode(char) {
+ return (char.charCodeAt(0) - 'A'.charCodeAt(0));
+}
+
+function getCodeChar(charValue) {
+ return (String.fromCharCode(('A'.charCodeAt(0) + charValue)));
+}
+
+function isNarsissticNumber(num) {
+ let sum = 0;
```

Comment:
Let's move away the Narcissistic Numbers logic into it's own scope using an IIFE and use a better OOPS pattern?